server/src/api/users/model.ts

```diff
@@ -0,0 +1,52 @@
+import { request } from 'express'
+import { Connection } from 'mysql2'
+
+import { toObj, lastInsertId } from '../../utils/db'
+
+export default class UserDAO {
+  constructor(private db: Connection) {}
+
+  public async createUser(username: string, password: string): Promise<number> {
+    let md5 = require('md5')
+    let hash = md5(password)
+    const user = (await this.db.execute(
+      `INSERT INTO Users(username, password_hash) VALUES (?, ?)`,
+      [username, hash]
+    )) as any
+    return await lastInsertId(this.db)
+  }
+
+  public async createSession(userId: number): Promise<string> {
+    const session = await this.db.execute(
+      `INSERT INTO Sessions(user_id, session_token) VALUES (?, UUID())`,
+      [userId]
+    )
+    const sessionId = await lastInsertId(this.db)
+    const result = await this.db
+      .promise()
+      .query('SELECT session_token FROM Sessions WHERE id = ?', [sessionId])
+    const rows = toObj(((result as unknown) as any)[0]) as any
+    return rows[0].session_token
+  }
+
+  public async loginUser(
+    username: string,
+    password: string
+  ): Promise<number | undefined> {
+    let md5 = require('md5')
```

Comment:
We could put this require at the top of the file instead so we don't re-require every request.

Reply:
So rather than being required in the loginUser function, would we just require the username and password at the class UserDAO?

Reply:
Ah, sorry, I meant at the the top of this file we can just put.
`const md5 = require('md5')`
That way we don't re-require it every request.